packages/sandbox/src/cloudwatch_logs_monitor.ts

```diff
@@ -0,0 +1,284 @@
+ * Consumers can activate and deactivate the monitor. The monitor on reactivation, starts
+ * streaming from the last time it was deactivated to avoid missing any logs while the monitor
+ * was deactivated.
+ */
+export class CloudWatchLogEventMonitor {
+  /**
+   * Determines from what time the logs should be streamed
+   */
+  private startTime: number;
+
+  /**
+   * Collection of all LogGroups that need to be streamed
+   */
+  private readonly allLogGroups: LogGroupStreamingCursor[] = [];
+
+  private readonly logGroupEventDisplay: Record<string, LogGroupEventDisplay> =
+    {};
+
+  private active = false;
+
+  private printer = printer; // default stdout
+
+  /**
+   * Initializes the start time to be `now`
+   */
+  constructor(readonly cloudWatchLogsClient: CloudWatchLogsClient) {
+    this.startTime = Date.now();
+  }
+
+  /**
+   * resume writing/printing events
+   */
+  activate = (): void => {
+    this.active = true;
+    this.scheduleNextTick(0);
+  };
+
+  /**
+   * deactivates the monitor so no new events are read
+   * use case for this is when we are in the middle of performing a deployment
+   * and don't want to interweave all the logs together with the CFN
+   * deployment logs
+   *
+   * Also resets the start time to be when the new deployment was triggered
+   * and clears the list of tracked log groups
+   */
+  deactivate = (): void => {
+    this.active = false;
+    this.startTime = Date.now();
+    this.allLogGroups.splice(0, this.allLogGroups.length);
+  };
+
+  /**
+   * Adds CloudWatch log groups to read log events from.
+   * Since we could be watching multiple logs groups we need a friendly
+   * name for to associate the log group to make it easier
+   * for the user to identify which log groups are being monitored
+   * @param friendlyResourceName The friendly name of the resource that is being monitored
+   * @param logGroupName The log group to read events from
+   */
+  addLogGroups = (friendlyResourceName: string, logGroupName: string): void => {
+    this.allLogGroups.push({
+      logGroupName,
+      startTime: this.startTime,
+    });
+    this.logGroupEventDisplay[logGroupName] = {
+      friendlyName: friendlyResourceName,
+      color: this.getNextColorForLogGroup(),
+    };
+  };
+
+  /**
+   * If output location file is specified, the logs will be appended to that file.
+   * If the file doesn't exist it will be created.
+   * @param outputLocation file location
+   */
+  setOutputLocation = (outputLocation: string) => {
+    if (!outputLocation) {
+      return;
+    }
+    const targetPath = path.isAbsolute(outputLocation)
+      ? outputLocation
+      : path.resolve(process.cwd(), outputLocation);
+    this.printer = new Printer(
+      LogLevel.INFO,
+      fs.createWriteStream(targetPath, { flags: 'a', autoClose: true })
+    );
+  };
+
+  /**
+   * Pick the next color in the object `colors` in round robin fashion
```

Comment:
Fwiw, as somebody who's partially colorblind I would encourage to add secondary disambiguation mechanism other than colors.
For example ability to prefix log lines with function name, something like:
```
[func1] log1
[func2] log2
```
and so on.

Reply:
yes that's how it is

```console
illed Duration: 69 ms   Memory Size: 512 MB     Max Memory Used: 90 MB
[defaultNodeFunction] 2:35:31 PM START RequestId: a8d1913a-598f-43e0-8a96-f7e566a63108 Version: $LATEST
[defaultNodeFunction] 2:35:32 PM END RequestId: a8d1913a-598f-43e0-8a96-f7e566a63108
[defaultNodeFunction] 2:35:32 PM REPORT RequestId: a8d1913a-598f-43e0-8a96-f7e566a63108 Duration: 1056.85 ms   Billed Duration: 1057 ms Memory Size: 512 MB     Max Memory Used: 90 MB
```

Reply:
Awesome, thanks.